EKrut-Server/src/ekrut/server/db/InventoryItemDAO.java

```diff
@@ -58,26 +112,25 @@ public InventoryItem fetchInventoryItem(int itemId, String ekrutLocation) {
 	}
 
 	/**
-	 * Fetch array list of InventoryItem object(s) from DB, identified by ekrutLocation.
+	 * Fetch array list of InventoryItem object(s) from DB, identified by
+	 * ekrutLocation.
 	 * 
-	 * @param ekrutLocation the unique machine identifier. 
-	 * @return	{@link ArrayList}&lt;{@link InventoryItem}&gt; list with all the inventory items assigned to the provided machine
+	 * @param ekrutLocation the unique machine identifier.
+	 * @return {@link ArrayList}&lt;{@link InventoryItem}&gt; list with all the
+	 *         inventory items assigned to the provided machine
 	 */
 	public ArrayList<InventoryItem> fetchAllItemsByLocation(String ekrutLocation) {
-		PreparedStatement ps = con.getPreparedStatement("SELECT * FROM inventory_items WHERE ekrutLocation = ?;");
+		PreparedStatement ps = con.getPreparedStatement("SELECT itemId FROM inventory_items WHERE ekrutLocation = ?;");
 		try {
 			ps.setString(1, ekrutLocation);
 			ResultSet rs = con.executeQuery(ps);
 			ArrayList<InventoryItem> inventoryItems = new ArrayList<>();
-			while(rs.next()) {
-				Item item = itemDAO.fetchItem(rs.getInt(1));
-				if (item != null)
-					inventoryItems.add(new InventoryItem(item, rs.getInt(2), rs.getString(3), rs.getInt(4)));
-			}
-			return inventoryItems.size() != 0? inventoryItems : null;
+			while (rs.next())
+				inventoryItems.add(fetchInventoryItem(rs.getInt("itemId"), ekrutLocation));
```

Comment:
This turns what could be accomplished with one database query into N database queries. See if you could maybe come up with a better way to do this that doesn't involve making unnecessary db accesses.